internal/topo/planner/logicalPlan.go

```diff
@@ -27,10 +35,58 @@ type LogicalPlan interface {
 	PruneColumns(fields []ast.Expr) error
 }
 
+type ExplainInfo interface {
+	ID() int64
+	Type() string
+	ChildrenID() []int64
+	Explain() string
+	BuildExplainInfo(id int64)
+}
+
+type PlanExplainInfo struct {
+	T        string  `json:"type"`
+	Info     string  `json:"info"`
+	Id       int64   `json:"id"`
+	Children []int64 `json:"children"`
+}
+
 type baseLogicalPlan struct {
 	children []LogicalPlan
 	// Can be used to return the derived instance from the base type
-	self LogicalPlan
+	self        LogicalPlan
+	ExplainInfo PlanExplainInfo
+}
+
+func (p *baseLogicalPlan) Explain() string {
+	p.ExplainInfo.T = p.Type()
+	p.ExplainInfo.Children = p.ChildrenID()
+	bf := bytes.NewBuffer([]byte{})
+	jsonEncoder := json.NewEncoder(bf)
+	jsonEncoder.SetEscapeHTML(false)
+	jsonEncoder.Encode(p.ExplainInfo)
+	return bf.String()
+}
+
+func (p *baseLogicalPlan) BuildExplainInfo(id int64) {
+	p.self.BuildExplainInfo(id)
+}
+
+func (p *baseLogicalPlan) Type() string {
+	ty := reflect.TypeOf(p.self)
+	s := strings.Split(ty.String(), ".")
+	return s[1]
```

Comment:
I don't think `reflect` is a proper way to get the type string, I tend to initialize the type string during init logical plan like tidb

https://github.com/pingcap/tidb/blob/6888a1aa6b0f31b1c23e28bc7811bcdf2c5e5ddf/planner/core/initialize.go#L41-L44